src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -488,4 +554,46 @@ private static string WritePgAdminServerJson(IEnumerable<PostgresServerResource>
 
         return filePath;
     }
+
+    private static async Task CreateDatabaseAsync(NpgsqlConnection npgsqlConnection, PostgresDatabaseResource npgsqlDatabase, IServiceProvider serviceProvider, CancellationToken cancellationToken)
+    {
+        var scriptAnnotation = npgsqlDatabase.Annotations.OfType<CreationScriptAnnotation>().LastOrDefault();
+
+        if (scriptAnnotation?.Script == null)
+        {
+            try
+            {
+                var quotedDatabaseIdentifier = new NpgsqlCommandBuilder().QuoteIdentifier(npgsqlDatabase.DatabaseName);
+                using var command = npgsqlConnection.CreateCommand();
+                command.CommandText = $"CREATE DATABASE {quotedDatabaseIdentifier}";
+                await command.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (PostgresException p) when (p.SqlState == "42P04")
+            {
+                // Ignore the error if the database already exists when Aspire tries to create it automatically.
+                // If the database was created by the user, then this exception would be thrown.
+            }
+            catch (Exception e)
+            {
+                var logger = serviceProvider.GetRequiredService<ResourceLoggerService>().GetLogger(npgsqlDatabase.Parent);
+                logger.LogError(e, "Failed to create database '{DatabaseName}'", npgsqlDatabase.DatabaseName);
+            }
+        }
+        else
+        {
+            // User-provided script
+
+            try
+            {
+                using var command = npgsqlConnection.CreateCommand();
+                command.CommandText = scriptAnnotation?.Script;
+                await command.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e)
```

Comment:
Don't we want to handle `            catch (PostgresException p) when (p.SqlState == "42P04")` here too? Or does the

Reply:
Since the script could contain other things than create database I didn't want to hide errors. But now it's only supporting create db so it's doing the same thing.